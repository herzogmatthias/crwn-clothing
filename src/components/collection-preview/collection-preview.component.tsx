import * as React from "react";
import { IShopItem } from "../../redux/shop/IShopItem";
import CollectionItem from "../collection-item/collection-item.component";
import {
  TitleContainer,
  CollectionPreviewContainer,
  PreviewContainer,
} from "./collection-preview.styles";
import { RouteComponentProps, withRouter } from "react-router-dom";

export interface ICollectionPreviewProps extends RouteComponentProps<{}> {
  title: string;
  items: IShopItem[];
  routeName: string;
}

export function CollectionPreview({
  title,
  items,
  routeName,
  history,
  match,
}: ICollectionPreviewProps) {
  return (
    <CollectionPreviewContainer>
      <TitleContainer
        onClick={() => history.push(`${match.path}/${routeName}`)}
      >
        {title.toUpperCase()}
      </TitleContainer>
      <PreviewContainer>
        {items
          .filter((itm, index) => index < 4)
          .map((item) => {
            return <CollectionItem key={item.id} item={item}></CollectionItem>;
          })}
      </PreviewContainer>
    </CollectionPreviewContainer>
  );
}

export default withRouter(CollectionPreview);
