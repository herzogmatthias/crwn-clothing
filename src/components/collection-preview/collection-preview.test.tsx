import React from "react";
import { shallow, ShallowWrapper } from "enzyme";

import { CollectionPreview } from "./collection-preview.component";
import { TitleContainer } from "./collection-preview.styles";

describe("NON CONNECTED COLLECTION-PREVIEW COMPONENT", () => {
  let wrapper: ShallowWrapper;
  let mockMatch: any;
  let mockHistory;
  const mockRouteName = "hats";

  beforeEach(() => {
    mockMatch = {
      path: "/shop",
    };

    mockHistory = {
      push: jest.fn(),
    };

    const mockProps = {
      match: mockMatch,
      history: mockHistory,
      routeName: mockRouteName,
      title: "hats",
      items: [],
      location: {},
    } as any;

    wrapper = shallow(<CollectionPreview {...mockProps} />);
  });

  test("render CollectionPreview component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("call history.push with the right string when TitleContainer clicked", () => {
    wrapper.find(TitleContainer).simulate("click");

    expect(mockHistory.push).toHaveBeenCalledWith(
      `${mockMatch.path}/${mockRouteName}`
    );
  });
});
