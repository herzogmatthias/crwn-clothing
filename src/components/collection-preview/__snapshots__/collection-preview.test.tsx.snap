// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NON CONNECTED COLLECTION-PREVIEW COMPONENT render CollectionPreview component 1`] = `
<styled.div>
  <styled.h1
    onClick={[Function]}
  >
    HATS
  </styled.h1>
  <styled.div />
</styled.div>
`;
